### ISSUE TITLE
As a funder, I want permission / reserved for UI on bounty new to be changed, so I can have a clean UX

### ISSUE THREAD

**Issue by owocki** [gitcoinco/web#3935]
### User Story

[comment]: # (As a <user type>, I want to <task> so that <goal>.)
As a funder, I want permission / reserved for UI on bounty new to be changed, so I can have a clean UX

### Why Is this Needed

[comment]: # (Describe the problem and why this task is needed. Provide description of the current state, what you would like to happen, and what actually happen)
*Summary*:
why is ‘permissions’ and ‘reserved for’ two seperate fields?

![slack-imgs](https://user-images.githubusercontent.com/513929/54046796-9c033f80-4192-11e9-9971-589def416bee.gif)

shouldnt ‘reserved for’ be a third option for permissions, and the reserved for field should show up only IFF it’s selected?

note my complaint isnt just about the UI.  the other day i reserved a bounty for someone, but i left ‘permissions’ as ‘approval required’ and then that person had to apply and i had to approve them down the line.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*:
Feature

### Current Behavior
[comment]: # (Describe what actually happened.)
Reserved For / Permissions are not coupled

### Expected Behavior
[comment]: # (Describe what you expected to happen.)
- ‘reserved for’ should be a third option for permissions, and the reserved for field should show up only IFF it’s selected
- any downstream effects of this in the start work application process should be QAed.

### Definition of Done
[comment]: # (Any other information that would be useful, bullets are helpful.)
see above

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 0.5 ETH (68.7 USD @ $137.41/ETH)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/gitcoinco/web/3935/2569).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/gitcoinco/web/3935/2569).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $39,961.07 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 12 months from now.
Please review their action plans below:


**1) [francopetra](https://gitcoin.co/profile/francopetra) has started work.**

Hi there! First issue I'm working on here, so I will get the repo on my local env. 
Will create a funder's account and emulate the situation on Rinkeby so I can test it out. (Not sure if will need this).
Change the UI and test. 
Create the PR to the repo.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/gitcoinco/web/3935/2569).

**Comment by francopetra** [comment]
Hi @PixelantDesign and @owocki! 
Have been moving forward with the UI part in [this branch](https://github.com/francopetra/web/tree/reserved-for
), changing the form behavior and some buttons.
I created two users and the workflow works (Funder selects user, and when the user requests access, s/he can start work). 

<img width="937" alt="Screen Shot 2019-03-12 at 9 12 57 AM" src="https://user-images.githubusercontent.com/20055787/54200977-4dfc7d80-44ab-11e9-9289-fa64e7e0c2af.png">

<img width="1183" alt="Screen Shot 2019-03-12 at 9 27 14 AM" src="https://user-images.githubusercontent.com/20055787/54200962-44731580-44ab-11e9-8971-1f84ff1c15e3.png">

**Still missing:**
* Change status on search from Ready to work, to Reserved
<img width="1145" alt="Screen Shot 2019-03-12 at 9 30 13 AM" src="https://user-images.githubusercontent.com/20055787/54200759-d9c1da00-44aa-11e9-9c59-332e0dcf2a7b.png">

* Change copy from reserved to Approved or working

* Change copy from "Express interest" to "Start working" or "Accept Bounty" 
<img width="1071" alt="Screen Shot 2019-03-12 at 9 30 56 AM" src="https://user-images.githubusercontent.com/20055787/54200817-feb64d00-44aa-11e9-8523-6ee371563313.png">

* Create a notification so the user knows someone has reserved a Bounty for them
No clue on how the notifications work on GitCoin. Any help here?

FInally thinking about edge cases. If the user stops working, the Permission type would continue to be "reserved". Should it change to Approval Required? If so, how? 

Would love some feedback!

**Comment by owocki** [comment]
this looks like really solid progress! 

for all copy youre waiting on, just stub some copy in, or put some lorem ipsum in there, and we'll modify in code review if needed.

>  If the user stops working, the Permission type would continue to be "reserved". Should it change to Approval Required? If so, how?

I think it should remain reserved, what do you think @PixelantDesign ?

**Comment by francopetra** [comment]
Thanks @owocki! (btw the project is amazing!), I will finish during the weekend. 
What do you think about the notifications? Is there anything else I can do from the front end? (Already sending "permission_type = reserved", which can help trigger the notif to the user from the backend.

**Comment by owocki** [comment]
i suppose someone should get a 'a bounty has been reserved for you' notification right?  but that was not part of the original scope.  @PixelantDesign what do you think?

**Comment by francopetra** [comment]
Hey @owocki  and @PixelantDesign just pushed the code for review. Was waiting for @PixelantDesign, but I also think it wasn't in the initial issue's scope, but it would be good for the contributor's workflow. 
I think the issue is pretty much solved. Let me know what you think!

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. **Submitted** 4. Done 

<hr>

__Work for 0.5 ETH (66.99 USD @ $133.97/ETH) has been submitted by__:

 1. [@francopetra](https://gitcoin.co/profile/francopetra)



@owocki please take a look at the submitted work:
* [PR](https://github.com/gitcoinco/web/pull/4036) by @francopetra

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/gitcoinco/web/3935/2569)
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/gitcoinco/web/3935/2569).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $51,238.21 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)
